src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/OpenDistroSecurityIndexSearcherWrapper.java

```diff
@@ -138,13 +138,17 @@ protected final boolean isBlockedIndexRequest() {
     protected final boolean isPermittedOnIndex() {
         final TransportAddress caller = (TransportAddress) this.threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_REMOTE_ADDRESS);
         final User user = (User) threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
-        if (caller == null || user == null) {
-            return false;
+
+        if (user != null && caller != null) {
+            final Set<String> securityRoles = evaluator.mapRoles(user, caller);
+            if (WildcardMatcher.matchAny(allowedRoles, securityRoles)) {
+                return true;
+            }
         }
-        final Set<String> securityRoles = evaluator.mapRoles(user, caller);
-        if (WildcardMatcher.matchAny(allowedRoles, securityRoles)) {
+        if ("true".equals(HeaderHelper.getSafeFromHeader(threadContext, ConfigConstants.OPENDISTRO_PROTECTED_INDICES_CONF_REQUEST_HEADER))) {
```

Comment:
We should probably do this only if "user == null", which means that only requests that don't have user context (originating from internal cron jobs) will have access.